### ISSUE TITLE
feat: proposal mechanism incentives 

### ISSUE THREAD

**Issue by Supremesource** [agicommies/subspace-network#96]
# Currently Governance Model is Not Sustainable

> This issue is still under construction.
> @rad2520 needs to edit.

## TL;DR

1. There is no incentive to participate (financial)
2. It requires too large amount of holder base to participate

We will focus on point 2) later, when we have proper subnet pricing mechanism, and subnet development has been bootstrapped.

## Solution to 1)

To solve the issue 1), since Commune already utilized the concept of DAO treasury, we should utilize its funds properly to motivate users in participating in the ecosystem.

### What We Do

**Storage Values:**

We define new global parameters:

- `GovernanceAllocation`: storage value, Percent, how many % of the treasury is allocated to governance (proposal participation)
- `MaxGovernanceAllocation`: storage value, u64, how many tokens can be allocated to the governance, example 100k
- `GovernanceAllocationInterval`: storage value, u16, how many blocks for the % allocation to renew

Feel free to suggest better naming.

**Logic Of Distribution:**

- How much everyone gets:
  When executing the proposal, we look at the stakes of the users voting on the proposal and adjust the allocation to be quadratic. (Large holders and small holders will be rewarded for governance participation).

- What is the value we are basing things off of:
  The amount that is going to get distributed corresponds to the storage values above. (Assuming this is not the first proposal, the second one will get the following allocation):

  First proposal:
  `allocation`

  Second proposal:
  `allocation / 2^2`

  Third proposal:
  `allocation / (2^2 * 2)`

  In general, for the n-th proposal:
  `allocation / 2^(2n)`
